nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/queue/GetAzureQueueStorage.java

```diff
@@ -0,0 +1,208 @@
+            context.yield();
+            return;
+        }
+
+        final List<CloudQueueMessage> cloudQueueMessages = toList(retrievedMessagesIterable);
+
+        for (final CloudQueueMessage message : cloudQueueMessages) {
+            FlowFile flowFile = session.create();
+
+            final Map<String, String> attributes = new HashMap<>();
+
+            attributes.put("azure.queue.uri", cloudQueue.getUri().toString());
+            attributes.put("azure.queue.insertionTime", message.getInsertionTime().toString());
+            attributes.put("azure.queue.expirationTime", message.getExpirationTime().toString());
+            attributes.put("azure.queue.messageId", message.getMessageId());
+            attributes.put("azure.queue.popReceipt", message.getPopReceipt());
+
+            flowFile = session.putAllAttributes(flowFile, attributes);
+            flowFile = session.write(flowFile, out -> {
+                try {
+                    out.write(message.getMessageContentAsByte());
+                } catch (StorageException e) {
+                    getLogger().error("Failed to write the retrieved queue message to FlowFile content due to {}", new Object[] {e});
+                    context.yield();
+                }
+            });
+
+            session.transfer(flowFile, REL_SUCCESS);
+            session.getProvenanceReporter().receive(flowFile, cloudQueue.getStorageUri().toString());
+        }
+
+        if(autoDelete) {
+            session.commit();
+
+            for (final CloudQueueMessage message : cloudQueueMessages) {
+                try {
+                    cloudQueue.deleteMessage(message);
+                } catch (StorageException e) {
+                    getLogger().error("Failed to delete the retrieved message with the id {} from the queue due to {}",
+                            new Object[] {message.getMessageId(), e});
+                }
+            }
+        }
+
+    }
+
+    @Override
+    public Collection<ValidationResult> customValidate(final ValidationContext validationContext) {
+        final List<ValidationResult> problems = new ArrayList<>(super.customValidate(validationContext));
+
+        final int batchSize = validationContext.getProperty(BATCH_SIZE).asInteger();
+
+        //Azure Queue Storage supports only a max of 32 messages to be retrieved in a single invocation
+        if (batchSize > 32) {
+            problems.add(new ValidationResult.Builder()
+                                             .valid(false)
+                                             .subject(BATCH_SIZE.getDisplayName())
+                                             .explanation("only up to 32 messages can be retrieved at a time")
+                                             .build()
+            );
+        }
+
+        final int visibilityTimeout = validationContext.getProperty(VISIBILITY_TIMEOUT).asTimePeriod(TimeUnit.SECONDS).intValue();
+
+        if (visibilityTimeout <= 0) {
```

Comment:
You can specify in the property that you want a positive integer validator instead of creating a custom validator

Reply:
My bad.. I'll fix it.

Reply:
No.. `VISIBILITY_TIMEOUT` is using `TIME_PERIOD_VALIDATOR` as expected.. this custom validation is to check if the validated time period (in seconds) is `0` or less. (less may not happen) The reason why it is checked: visibility timeout of 0 secs will throw an error in runtime.. Not sure if that's a bug in the storage SDK.